src/main/java/de/zalando/aruha/nakadi/util/JsonPathAccess.java

```diff
@@ -0,0 +1,90 @@
+package de.zalando.aruha.nakadi.util;
+
+import de.zalando.aruha.nakadi.exceptions.InvalidOrderingKeyFieldsException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/*
+ One could use JsonPath Lib instead: https://github.com/jayway/JsonPath
+
+ However, I had the feeling that JsonPath is way too much compared to what is needed here. Therefore it might be
+ slower. Moreover we already have a JSONObject. Using JsonPath would mean to convert it back to a string (or pass
+ the json string instead of the JSONObject to the strategy) and parse it again.
+ */
+public class JsonPathAccess {
+    private final JSONObject jsonObject;
+
+    public JsonPathAccess(final JSONObject jsonObject) {
+        this.jsonObject = jsonObject;
+    }
+
+    public Object get(final String path) throws InvalidOrderingKeyFieldsException {
+
+        final JsonPathTokenizer pathTokenizer = new JsonPathTokenizer(path);
+
+        Object curr = this.jsonObject;
+        String field;
+
+        while ((field = pathTokenizer.nextToken()) != null) {
+            if (!(curr instanceof JSONObject)) {
+                throw new InvalidOrderingKeyFieldsException("field " + field + "doesn't exist.");
+            }
+            try {
+                curr = ((JSONObject) curr).get(field);
+            } catch (JSONException e) {
+                throw new InvalidOrderingKeyFieldsException("field " + field + "doesn't exist.");
+            }
+        }
+        return curr;
+    }
+
+    private static class JsonPathTokenizer {
+        private final char[] path;
+        private int pos = 0;
+
+        private JsonPathTokenizer(final String path) {
+            this.path = path.toCharArray();
+        }
+
+        public synchronized String nextToken() {
```

Comment:
Does it really needs synchronization? If not, then tokenBuilder object could be created only once.
But on every iteration it needs to be reset by calling `setLength(0)`